src/Rendering_Internals.md

```diff
@@ -0,0 +1,408 @@
+If the tile is flipped horizontally (attributes bit 5), the byte read is flipped around.
+(The timing of horizontal flipping has not been verified.)
+
+This step takes 2 dots.
+
+### Get tile row (high)
+
+Exactly the same as [Get tile slice (low)](<#Get tile row (low)>), except the following byte is fetched (i.e. bit 0 of the address is 1 instead of 0).
+This step takes 2 dots as well.
+
+#### Bitplane desync
+
+Interesting phenomena can be triggered by changing the address' "parameters" between the two bitplane reads, called "bitplane desyncing".
+Since VRAM and OAM cannot be modified during Mode 3 (though OAM DMA can change what the PPU reads from OAM), the parameters that can be changed are [`SCY`](<#FF42–FF43 — SCY, SCX: Viewport Y position, X position>) and [`LCDC bit 4`](<#LCDC.4 — BG and Window tile data area>).
+
+Modifying `SCY` causes the second bitplane (and also the first one, depending on timing) to be read from a different Y offset within the tile than normal.
+This does not occur starting with CGB revision D, including AGBs: `SCY` is internally latched during the tilemap read, so both bitplanes are always read correctly.
+(Compare [CGB-C](https://github.com/mattcurrie/mealybug-tearoom-tests/blob/70e88fb90b59d19dfbb9c3ac36c64105202bb1f4/expected/CPU%20CGB%20C/m3_scy_change.png) and [CGB-D](https://github.com/mattcurrie/mealybug-tearoom-tests/blob/master/expected/CPU%20CGB%20D/m3_scy_change.png).)
+
+Modifying `LCDC` bit 4 exhibits much more complex behavior, [explained in this document by mattcurrie](https://github.com/mattcurrie/mealybug-tearoom-tests/blob/70e88fb90b59d19dfbb9c3ac36c64105202bb1f4/the-comprehensive-game-boy-ppu-documentation.md#tile_sel-bit-4).
+
+### Push pixels
+
+Once the fetcher reaches this state, it will attempt to push the two bytes it read, plus associated metadata, into the target FIFO on every dot.
+
+The BG FIFO will only accept pixels when it's empty.
+
+The OBJ FIFO needs to "merge" the pixels being pushed with the pixels it already contains.
+This decision leads to what is conceptually known as ["OBJ-to-OBJ priority"](<#Drawing priority>).
+Note that overwriting a pixel entails not only replacing its ID, but also any attached attributes, such as the palette, etc.
+
+- On DMG and on CGB in Non-CGB Mode, the algorithm is simply that only pixels with an ID of 0 (= transparent pixels) are overwritten.
+  Since OBJ pixels are inserted as the OBJs are encountered horizontally, pixels inserted earlier, thus from objects with lower X positions, will have priority.
+- On CGB in CGB Mode, OBJ pixels also store the ID of the OBJ they originated from, and get overwritten by pixels from OBJs with lower IDs.
+
+## Selector
+
+Every time both FIFOs are clocked, the selector decides whether to retain the pixel from the BG or OBJ FIFO.
+
+The selection follows the following rules:
+1. **In CGB Mode**, if [`LCDC` bit 0 (priority enable)](<#CGB Mode: BG and Window master priority>) is reset, pick the BG pixel.
```

Comment:
This rule appears to be simply disabling OBJ, which duplicates the functionality of rule 3 and doesn't seem correct. I believe a correct rule would be to ignore rules 4 and 5.